linkerd/service-profiles/src/lib.rs

```diff
@@ -105,3 +112,59 @@ fn stream_profile(mut rx: Receiver) -> Pin<Box<dyn Stream<Item = Profile> + Send
         }
     })
 }
+
+// === impl LookupAddr ===
+
+impl fmt::Display for LookupAddr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl FromStr for LookupAddr {
+    type Err = <Addr as FromStr>::Err;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Addr::from_str(s).map(LookupAddr)
+    }
+}
+
+impl From<Addr> for LookupAddr {
+    fn from(a: Addr) -> Self {
+        Self(a)
+    }
+}
+
+impl Into<Addr> for LookupAddr {
+    fn into(self) -> Addr {
+        self.0
+    }
+}
+
+// === impl LogicalAddr ===
+
+impl fmt::Display for LogicalAddr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl FromStr for LogicalAddr {
+    type Err = <NameAddr as FromStr>::Err;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        NameAddr::from_str(s).map(LogicalAddr)
+    }
+}
+
+impl From<NameAddr> for LogicalAddr {
+    fn from(na: NameAddr) -> Self {
+        Self(na)
+    }
+}
+
+impl Into<NameAddr> for LogicalAddr {
+    fn into(self) -> NameAddr {
+        self.0
+    }
+}
```

Comment:
a lot of these are pretty trivial, maybe they should be marked as `#[inline]`able? not sure if this matters at all, just noticed that we added the attribute elsewhere in this branch...